### ISSUE TITLE
SIM109—potential false positive

### ISSUE THREAD

**Issue by patrick91** [charliermarsh/ruff#1993]
Hi there, I have this code this:

```python
import abc

def x(annotation_origin):
    return (
        annotation_origin == list
        or annotation_origin == tuple
        or annotation_origin is abc.Sequence
    )
```

that gets converted to

```python
import abc

def x(annotation_origin):
    return (
        annotation_origin in (list, tuple)
    )
```

I'm not 100% sure this is a bug, but the previous code we had is valid. The same behaviour happens with Flake8

[Playground link](https://ruff.pages.dev/#N4KABGBECGA2sHsDuBTAJgWgMYIHYDMBXAZ2gCNYVjIAuMAbQF0AacKMwgS1gBdPdqdJqwiQ0hALYSAnhgBu0AE6dylDIoDmAD1pQAegAoA+gGoAPsZP1oGAF4BBDAC0ADBgCcRxgCprdx64ejCYA-ACUYQAkkCJQKFo8KLiYnBq4CIoousJskPGJyRjEKJRYPNksufkoirhwFbGQqemZDbmw-CgYlLgaPAAWugAcQ43FpeVCbKIAojHTUABi8xCiAMoAkgCyAIwu7pBslaI8ShooPPI1xJx4upAADtIAzHsrUPiw0ADWKEMY0Fw6VOfDwgjAoFWUBkTww-E4l0yPEItV0+DgxViomIhAeD0yxGIGHEUlkZ3B6NgmIWkBxeIJRPSuC6SJRuH4GjRGJQWKgcEQSCKp0UAM0ANw0i5VJQbAAvo1Pj8-hgyIC0AjdJDVpB+ko0Dg0OgMDwJA9iZxMmUMpKplDRAB6E0Pd5QyD2hSKR2ml3a+2GuT24j9CSHKGMOUKr6-f4cDRkFBKTU06qFThSQinChdLD88FMCO5RXRjA1RQSYicuha0QSaBaIXKXrdJJ9QZ0FwF0RF5UAR0ICES4OrUH4HWZGD7A6o9zQCEIWZ9kAkhF43E6E-7g5nc4XvLECCwxB4jY0G6n4P388oi+gcgQnEwVBzDyydGPhBlEHlhajyr4aFkNMHgyHghxpVVcHUEpoD4OQuiAkCLweJQklAxcIOZTBoAeThNVlTsPl-f4EMUS4cFwODcFBAQk1dOAVGKMC7SgZDkmgJC0B9bEEzIDJcHuYgaN5URcEkJ57lwZ1hL5XhoAte44HKaSlxgh5EB4DoyAAOieNSB3uNTygWfCv0jJV-ieQdLiPaRryrGl8E4BIUS6ZDMio-oqGnN9FA-Pc3OgCQLmUWwujqIKiR4aQX3uZE1KyfylEC4LOFC+Q4A-SLotfKAOiPRcAqC49Uq6BRYEy9RkGNbLYtxa890UOTGKgvsLS6WseCwfoMHwDJsgWUQACF2JQGYtCwFAHmorioDGiaptufjlIANQy0bFEUPrlIAeTWGYNq2gaoA2bb9s2xQZsgGYKItPAgqos7DuY2kD1+HgtNLPqFmObVGs4ZqU0wTJWsyDAOq6nq+qEH6ayUb4MDc1DPMY3R30-MBvy7IiMEIUTiiwzRJFQpjtWaDJSqUFR1SwDBwu8sBKWKAimmIEDaO1XrFAmjAkEas16JGiluT3TnuZuXo1DHHLGZ5GlxY0SX13iLBypuME2ldHAJDIdd2LhU1EKlalXWINSEQwPBjT+sdTy12tUd82XXQyQ0RTIWQopiny-IcjIxZAnmEX6fgihQMpFqF6U92+dIkEgxzFCPBGlCijXtRj5BIIGC1MDc1Pob3fJGqFNU9W6TgyEaxQbQYGGoEyL5YPgg3SKJF2anuIhSORy4eAQbBEEYpSaWBrhMhSFu0ILmkVfYlHp81sFTiovM68gD0VCzVe93ScvmSJeNOZy-NTNyCQsBzeN2dh+stfirQERrvZmZfB5-nCjlr5HNJydp5K8yOrSC4ABVKSgDEhKAACKZ0usUHgoCADCXxCSXQgYoaBcckFz1gSAh4WwEDiHquAhM6DM74MIQlQB7FpC4CwGsXBl1qG0IACokIwUtQBcDQGsKPJAmC0BLrom4C5Oak1prKUQL0LYVBSDnEurWLQkDOD4HwKGVYa9Z6EiKv0AhxIw4ZBghkABz1NHEG0QQtREA15HhgpwLA5jMCGhwI1Puid+rPRsXwexFwdGcW+i-aQBoqBRTshCGkZNQYIDgooSRp5TjEHhnbYmRt0aYxYtIWch4Qk5WAGkx4tl+CTDCXReAVVawaDselcqXRPb03oJw48liwDhlPqIJ4uINCNUNF-SAvxJrqFxnwIK1UcK9BSXKEAspIjTJIjwMA5AsAgBAIafAYAtAGEBMCWxeAjAZFSPwMINAFislqGAAwR0ICbIHNs3AuzlAVNwGAAAvE8sAeUeAXLABkeZQJrnUTufsx5LywBxUoJ875VyQSLQBQ8sA-15lkCwFpehfYkgTQWGEEAQA)

**Comment by charliermarsh** [comment]
It looks like a bug to me.

**Comment by patrick91** [comment]
By the way, my “workaround” was:

```python
import abc

def x(annotation_origin):
    return (
        annotation_origin in (list, tuple)  
    ) or annotation_origin is abc.Sequence
```
